src/program/program.py

```diff
@@ -44,6 +42,7 @@ class Program(threading.Thread):
     def __init__(self):
         super().__init__(name="Riven")
         self.initialized = False
+        self.running = False
```

Comment:
Whats this?

Reply:
Something threw an error because `running` was missing from program. I'll have to track it back down again as that was like 2 days ago :/

Reply:
Ah it was program.stop

```py    
def stop(self):
    if not self.running:
        return

    self.running = False
    if hasattr(self, "executors"):
        for executor in self.executors:
            if not executor["_executor"]._shutdown:
                executor["_executor"].shutdown(wait=False)
    if hasattr(self, "scheduler") and self.scheduler.running:
        self.scheduler.shutdown(wait=False)
    logger.log("PROGRAM", "Riven has been stopped.")
```